pkg/remote/fileshare/wavefs/wavefs.go

```
@@ -346,18 +371,111 @@ func (c WaveClient) AppendFile(ctx context.Context, conn *connparse.Connection,
+func (c WaveClient) CopyRemote(ctx context.Context, srcConn, destConn *connparse.Connection, srcClient fstype.FileShareClient, opts *wshrpc.FileCopyOpts) error {
+	zoneId := destConn.Host
+	if zoneId == "" {
+		return fmt.Errorf("zoneid not found in connection")
+	}
+	readCtx, cancel := context.WithCancelCause(ctx)
+	ioch := srcClient.ReadTarStream(readCtx, srcConn, opts)
+	err := tarcopy.TarCopyDest(readCtx, cancel, ioch, func(next *tar.Header, reader *tar.Reader) error {
+		if next.Typeflag == tar.TypeDir {
+			return nil
+		}
+		fileName, err := cleanPath(path.Join(destConn.Path, next.Name))
```

golangci-lint (1.62.2)

ineffectual assignment to err

(ineffassign)